src/config.rs

```diff
@@ -1,46 +1,50 @@
 use crate::subprocess::SubProcess;
 use anyhow::{Context, Result};
-use serde::{Deserialize, Serialize};
+use serde::Deserialize;
 use std::{fs, path::PathBuf};
 
-#[derive(Serialize, Deserialize)]
+#[derive(Debug, Default, Deserialize, Eq, PartialEq)]
 pub struct Config {
+    #[serde(default)]
+    pub temporary_project_dir: Option<PathBuf>,
     #[serde(default)]
     pub welcome_message: bool,
     #[serde(default)]
-    pub cargo_target_dir: Option<PathBuf>,
+    pub prompt: bool,
     #[serde(default)]
     pub preserved_project_dir: Option<PathBuf>,
     #[serde(default)]
-    pub prompt: bool,
-    #[serde(default)]
-    pub editor: Option<String>,
+    pub vcs: Option<String>,
     #[serde(default)]
-    pub editor_args: Option<Vec<String>>,
+    pub git_repo_depth: Option<Depth>,
     #[serde(default)]
-    pub temporary_project_dir: Option<PathBuf>,
+    pub cargo_target_dir: Option<PathBuf>,
     #[serde(default)]
-    pub git_repo_depth: Option<Depth>,
+    pub editor: Option<String>,
     #[serde(default)]
-    pub vcs: Option<String>,
+    pub editor_args: Option<Vec<String>>,
     #[serde(default, rename = "subprocess", skip_serializing_if = "Vec::is_empty")]
     pub subprocesses: Vec<SubProcess>,
 }
 
 impl Config {
-    fn new() -> Result<Self> {
-        Ok(Self {
-            welcome_message: true,
-            cargo_target_dir: None,
-            preserved_project_dir: None,
-            prompt: false,
-            editor: None,
-            editor_args: None,
-            git_repo_depth: None,
-            temporary_project_dir: Some(Self::default_temporary_project_dir()?),
-            vcs: None,
-            subprocesses: Default::default(),
-        })
+    fn template() -> Result<String> {
+        let temporary_project_dir = Config::default_temporary_project_dir()?
+            .to_str()
+            .expect("path shouldn't contains invalid unicode")
+            .replace('\\', "\\\\");
+
+        #[cfg(target_family = "windows")]
+        let editor = "C:\\\\Program Files\\\\Microsoft VS Code\\\\Code.exe";
+        #[cfg(target_family = "unix")]
+        let editor = "/usr/bin/code";
```

Comment:
On Linux it's probably best to just put "code" here because you don't know where the editor will be installed.

It's possible on Windows that Code.exe is in the PATH but I don't know for sure. We need to ask someone who uses the Windows OS.